gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/BigQueryImplTest.java

```diff
@@ -0,0 +1,982 @@
+      TableInfo.of(TABLE_ID_WITH_PROJECT, TABLE_SCHEMA);
+  private static final JobInfo LOAD_JOB = LoadJobInfo.of(TABLE_ID, "URI");
+  private static final JobInfo LOAD_JOB_WITH_PROJECT = LoadJobInfo.of(TABLE_ID_WITH_PROJECT, "URI");
+  private static final JobInfo COMPLETE_LOAD_JOB = LoadJobInfo.builder(TABLE_ID_WITH_PROJECT, "URI")
+      .jobId(JobId.of(PROJECT, JOB))
+      .build();
+  private static final JobInfo COPY_JOB =
+      CopyJobInfo.of(TABLE_ID, ImmutableList.of(TABLE_ID, TABLE_ID));
+  private static final JobInfo COPY_JOB_WITH_PROJECT =
+      CopyJobInfo.of(TABLE_ID_WITH_PROJECT, ImmutableList.of(TABLE_ID_WITH_PROJECT,
+          TABLE_ID_WITH_PROJECT));
+  private static final JobInfo COMPLETE_COPY_JOB =
+      CopyJobInfo.builder(TABLE_ID_WITH_PROJECT, ImmutableList.of(TABLE_ID_WITH_PROJECT,
+          TABLE_ID_WITH_PROJECT))
+          .jobId(JobId.of(PROJECT, JOB))
+          .build();
+  private static final JobInfo QUERY_JOB = QueryJobInfo.builder("SQL")
+      .defaultDataset(DatasetId.of(DATASET))
+      .destinationTable(TABLE_ID)
+      .build();
+  private static final JobInfo QUERY_JOB_WITH_PROJECT = QueryJobInfo.builder("SQL")
+      .defaultDataset(DatasetId.of(PROJECT, DATASET))
+      .destinationTable(TABLE_ID_WITH_PROJECT)
+      .build();
+  private static final JobInfo COMPLETE_QUERY_JOB = QueryJobInfo.builder("SQL")
+      .defaultDataset(DatasetId.of(PROJECT, DATASET)).destinationTable(TABLE_ID_WITH_PROJECT)
+      .jobId(JobId.of(PROJECT, JOB))
+      .build();
+  private static final JobInfo EXTRACT_JOB = ExtractJobInfo.of(TABLE_ID, "URI");
+  private static final JobInfo EXTRACT_JOB_WITH_PROJECT =
+      ExtractJobInfo.of(TABLE_ID_WITH_PROJECT, "URI");
+  private static final JobInfo COMPLETE_EXTRACT_JOB =
+      ExtractJobInfo.builder(TABLE_ID_WITH_PROJECT, "URI")
+          .jobId(JobId.of(PROJECT, JOB))
+          .build();
+  private static final TableCell BOOLEAN_FIELD = new TableCell().setV("false");
+  private static final TableCell INTEGER_FIELD = new TableCell().setV("1");
+  private static final TableRow TABLE_ROW =
+      new TableRow().setF(ImmutableList.of(BOOLEAN_FIELD, INTEGER_FIELD));
+  private static final QueryRequest QUERY_REQUEST = QueryRequest.builder("SQL")
+      .maxResults(42L)
+      .useQueryCache(false)
+      .defaultDataset(DatasetId.of(DATASET))
+      .build();
+  private static final QueryRequest QUERY_REQUEST_WITH_PROJECT = QueryRequest.builder("SQL")
+      .maxResults(42L)
+      .useQueryCache(false)
+      .defaultDataset(DatasetId.of(PROJECT, DATASET))
+      .build();
+
+  // Empty BigQueryRpc options
+  private static final Map<BigQueryRpc.Option, ?> EMPTY_RPC_OPTIONS = ImmutableMap.of();
+
+  // Dataset options
+  private static final BigQuery.DatasetOption DATASET_OPTION_FIELDS =
+      BigQuery.DatasetOption.fields(BigQuery.DatasetField.ACCESS, BigQuery.DatasetField.ETAG);
+
+  // Dataset list options
```

Comment:
We could add the "max results" option here and set it to something low for testing purposes